src/prefect_server/graphql/states.py

```diff
@@ -23,6 +23,7 @@ async def resolve_set_flow_run_states(
 
     server_context = context.get_context()
     agent_id = server_context.get("headers", {}).get("x-prefect-agent-id")
+    flow_concurrency_lock = server_context.get("flow_concurrency_lock")
```

Comment:
We can remove this and have the `api.states.set_flow_run_state` call directly pull this lock from the server context. I had originally added it here as we're also pulling `agent_id` from the context, so I wanted to group those together.

One downside of this (and only passing it as a parameter instead of reading from the server context in the function directly) is that we'd only be able to test the limiting behavior via the `graphql` tests and not the `api` tests.